modules/check_in/spec/services/v2/chip/service_spec.rb

```diff
@@ -337,4 +340,185 @@
+                emergencyContactConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when only demographics_up_to_date in check_in_body' do
+          let(:params) do
+            {
+              demographics_up_to_date: true
+            }
+          end
+
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {
+                demographicsNeedsUpdate: false,
+                demographicsConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation with only demographics_up_to_date data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when only next_of_kin_up_to_date in check_in_body' do
+          let(:params) do
+            {
+              next_of_kin_up_to_date: false
+            }
+          end
+
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {
+                nextOfKinNeedsUpdate: true,
+                nextOfKinConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation with only next_of_kin_up_to_date data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when only emergency_contact_up_to_date in check_in_body' do
+          let(:params) do
+            {
+              emergency_contact_up_to_date: true
+            }
+          end
+
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {
+                emergencyContactNeedsUpdate: false,
+                emergencyContactConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation with only emergency_contact_up_to_date data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
+
+        context 'when no demographics data available in check_in_body' do
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {}
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
```

Comment:
Shouldn't this say "... with **no** demographics data"?

Reply:
Thank you, copy-paste issue.  Fixed it [604e461](https://github.com/department-of-veterans-affairs/vets-api/pull/9336/commits/604e4613bd3bd2d414058dde830ed8cb55b7688f)